Allow creation of N2 diagrams for problems that don't exist on global…

A-CGray opened the pull request:
### Summary

Allows the n2 diagram creator to be run for problems that do not exist on the global root proc by running on the root of the problem's comm rather than the root of the global comm.

### Related Issues

- Resolves #3077 

### Backwards incompatibilities

None

### New Dependencies

None

swryan commented:
would you be able to put together a small unit test to demonstrate your use case and the correct functionality per your fix?

A-CGray commented:
> would you be able to put together a small unit test to demonstrate your use case and the correct functionality per your fix?

Sure, I added a test that creates 2 problems on two comms and checks that they both write out files. I also added an assert to check the output file exists to the existing `N2ParallelTestCase` because it seemed like all that test was doing was checking that the call to `n2` doesn't hang. Feel free to remove if you want.

swryan commented:
Note: The 'Ubuntu Minimal` will fail, just ignore that.  we have a fix on another PR.
